### DISCUSSION TITLE
How to design an approval scenario?

### DISCUSSION THREAD

**Discussion by ekansh005** [Budibase/budibase#3898] [Help]
Hi,

I am relatively new to Budibase but trying to evaluate for one of my requirements i.e.
1. User creates a 'Distributor' via a form 
2. It goes for multi-level of approvals 
3. Once final level of approval is done, we trigger an API and pass the data

I tried to implement this but hitting road block time and again.
For the sake of simplicity, we can assume only one level of approval with hardcoded approver email id. But the design should support multi-level of approvals. The main approach that I opted using Budibase DB is:
1. One table for Distributors, another table for Approval Tasks. I need to establish a relation between distributor and approval task. 
2. Once a new distributor is created, automation triggers (row created in distributor table) and 
- update distributor status as 'Pending Approval' (not implemented yet)
- try to put a new row in approval tasks table.  The problem here is that in create new row step, I am not able to pass the name of distributor as it is a dropdown showing existing records in distributor table whereas I need to bind with the value coming from trigger step.

To be implemented once we solve the above issues 
3. Send an email to approver with a link to approval task page.
4. Approver opens the approval task from the link or from top navigation area where his/her pending approvals are shown and approves/rejects the distributor.
5. The distributor status is set to approved/rejected.
6. Webhook is triggered through automation trigger update row in distributor table.

Can you please help me in designing this in Budibase?

**Comment by ekansh005** [comment]
Hi,
I understand that holiday season has just ended but do you have any suggestion to design this scenario?

BR, Ekansh

**Comment by mjashanks** [comment]
Hi Ekansh.

> The problem here is that in create new row step, I am not able to pass the name of distributor...

This issue will be fixed by the end of this month.

However, I would take a more simple approach to approval workflows. I would not use an "approval task", but just base it all on the distributor status (unless there is another reason to have that table).

When the user first create's the record, we can set the status to a particular  status - lets say "Awaiting Approval", using the `Save Row` button action (i.e. choose "Define Actions" on a button)...

![image](https://user-images.githubusercontent.com/3524181/148121496-99afbf6f-1bac-47db-b474-7d6f01c745fe.png)

I would then remove the Status dropdown from the form, so that "Awaiting Approval" gets set automatically.

I can do the save at various stages of the flow. For example, I could create an "Approve" and "Reject" button that only updated the status.

I would then have separate screens for my different user types. For example "Approvers" would see a screen that listed records that were "Awaiting Approval".

hope this helps!
